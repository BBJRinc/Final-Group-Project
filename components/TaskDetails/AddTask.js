import React, { Component } from 'react';
import { Modal, TouchableHighlight, StatusBar } from 'react-native';
import { View, Input, Header, Right, Left, Content, Title, Button, Item, Body } from 'native-base';
import IconI from 'react-native-vector-icons/Ionicons';
import axios from 'axios';

<<<<<<< HEAD
const PubIpAddress = '192.168.3.176'
=======
const PubIpAddress = '192.168.3.132';
>>>>>>> master

export default class AddTask extends Component {

    constructor(props) {
        super(props)
        this.state = {
            taskname: '',
        }
    }

    handleTaskName(text) {
        this.setState({ taskname: text })
    }
    saveTaskName() {
        axios({
            method: 'post',
            url: `http://${PubIpAddress}:4040/api/task`,
            headers: {
                "token": this.props.token
            },
            data: {
                taskname: this.state.taskname,
            }
        }).then((res) => {
            this.props.setSelectedTask(res.data)
            this.props.showMenuItem('showAddTask');
            this.props.showMenuItem('showTaskDetails');
        })
    }
    render() {
        const { contentView, inputStyle } = style
        return (
                <Modal
                    style={contentView}
                    animationType="slide"
                    transparent={true}
                    visible={this.props.visible}
                    onRequestClose={() => this.props.showMenuItem('showAddTask')} >
                    <Header style={{ backgroundColor: '#000' }}>
                        <StatusBar barStyle='light-content' />
                        <Left>

                        </Left>
                        <Body><Title style={{ color: '#fff' }}>Add Task</Title></Body>
                        <Right>
                            <Button transparent onPress={() => this.props.showMenuItem('showAddTask')}>
                                <IconI name='ios-close' color='#fff' size={20} />
                            </Button>
                        </Right>
                    </Header>
                    <Content style={{ backgroundColor: '#00000099' }}>
                        <Item style={[contentView, { borderWidth: 1, borderBottomColor: '#fff' }]}>
                            <Input placeholder={'click to add a new task'} style={inputStyle} onChangeText={(text) => this.handleTaskName(text)} onEndEditing={() => this.saveTaskName()} />
                        </Item>
                    </Content>
                </Modal>
        )
    }
}
style = {
    contentView: {
        justifyContent: 'center',
        alignContent: 'center',
        alignItems: 'center',
        backgroundColor: (225, 225, 225, .8),
        marginTop: 225
    },
    inputStyle: {
        borderWidth: 1,
        borderColor: 'gray',
        backgroundColor: 'black',
        color: 'white',
        height: 40
    }
}