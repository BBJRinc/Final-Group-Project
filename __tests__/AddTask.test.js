// Brandons tests

import React, { Component } from 'react';
import AddTask from '../components/TaskDetails/AddTask';


import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<AddTask />).toJSON();
    expect(tree).toMatchSnapshot();
});

const BFunctions = require('../src/utils/BFunctions')
it('should change value', () => {
    let handleChange = renderer.create(<AddTask />).getInstance();
    handleChange.handleTaskName('some_text')
})
