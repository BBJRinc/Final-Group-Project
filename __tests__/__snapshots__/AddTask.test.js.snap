// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  style={
    Object {
      "alignContent": "center",
      "alignItems": "center",
      "backgroundColor": 0.8,
      "justifyContent": "center",
      "marginTop": 225,
    }
  }
  transparent={true}
  visible={true}
>
  <View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 15,
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "#000",
          },
        ]
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "600",
                "textAlign": "center",
              },
              Object {
                "color": "#fff",
              },
            ]
          }
          virtual={undefined}
        >
          Add Task
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 50,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginRight": -8,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 20,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            ÔêÜ
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={undefined}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
        Object {
          "backgroundColor": "#00000099",
        },
      ]
    }
    viewIsInsideTabBar={false}
    virtual={undefined}
  >
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": 0.8,
            "borderBottomColor": "#fff",
            "borderBottomWidth": 1,
            "borderColor": "#D9D5DC",
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginLeft": 2,
            "marginTop": 225,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <TextInput
          allowFontScaling={true}
          editable={true}
          onChangeText={[Function]}
          onEndEditing={[Function]}
          placeholder="click to add a new task"
          placeholderTextColor="#575757"
          style={
            Array [
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 50,
                "lineHeight": 24,
                "paddingLeft": 5,
                "paddingRight": 5,
                "top": 1.5,
              },
              Object {
                "backgroundColor": "black",
                "borderColor": "gray",
                "borderWidth": 1,
                "color": "white",
                "height": 40,
              },
            ]
          }
          underlineColorAndroid="rgba(0,0,0,0)"
          virtual={undefined}
        />
      </View>
    </View>
  </RCTScrollView>
</Modal>
`;
